Content.Client/_Goobstation/Fishing/Overlays/FishingOverlay.cs

```diff
@@ -0,0 +1,137 @@
+// SPDX-FileCopyrightText: 2025 Aidenkrz <aiden@djkraz.com>
+// SPDX-FileCopyrightText: 2025 GoobBot <uristmchands@proton.me>
+// SPDX-FileCopyrightText: 2025 Roudenn <romabond091@gmail.com>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+using System.Numerics;
+using Content.Client.UserInterface.Systems;
+using Content.Goobstation.Shared.Fishing.Components;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Shared.Enums;
+using Robust.Client.Player;
+using Robust.Shared.Utility;
+
+namespace Content.Goobstation.Client.Fishing.Overlays;
+
+public sealed class FishingOverlay : Overlay
+{
+    private readonly IEntityManager _entManager;
+    private readonly IPlayerManager _player;
+    private readonly SharedTransformSystem _transform;
+    private readonly ProgressColorSystem _progressColor;
+	private readonly SpriteSystem _sprite;
+    private readonly Texture _barTexture;
+
+    // Fractional positions for progress bar fill (relative to texture height/width)
+    private const float StartYFraction = 0.09375f; // 3/32
+    private const float EndYFraction = 0.90625f; // 29/32
+    private const float BarWidthFraction = 0.2f; // 2/10
+
+    // Apply a custom scale factor to reduce the size of the progress bar
+    // We dont want to do this because muh pixel consistency, but i'll keep it here as an option
+    private const float BarScale = 1f;
+
+    public override OverlaySpace Space => OverlaySpace.WorldSpaceBelowFOV;
+
+    public FishingOverlay(IEntityManager entManager, IPlayerManager player)
+    {
+        _entManager = entManager;
+        _player = player;
+        _transform = _entManager.EntitySysManager.GetEntitySystem<SharedTransformSystem>();
+        _progressColor = _entManager.System<ProgressColorSystem>();
+		_sprite = _entManager.System<SpriteSystem>();
```

Comment:
Spaces instead of tabs
```suggestion
        _sprite = _entManager.System<SpriteSystem>();
```